setup.py

```diff
@@ -87,19 +69,27 @@ def _load_py_module(name: str, location: str) -> ModuleType:
     return py
 
 
-# https://packaging.python.org/discussions/install-requires-vs-requirements /
-# keep the meta-data here for simplicity in reading this file... it's not obvious
-# what happens and to non-engineers they won't know to look in init ...
-# the goal of the project is simplicity for researchers, don't want to add too much
-# engineer specific practices
 if __name__ == "__main__":
-    _SETUP_TOOLS = _load_py_module(name="setup_tools", location=os.path.join(".actions", "setup_tools.py"))
-
-    if _PACKAGE_NAME not in _PACKAGE_MAPPING:  # install everything
-        _SETUP_TOOLS._load_aggregate_requirements(_PATH_REQUIRE, _FREEZE_REQUIREMENTS)
-
-    _SETUP_TOOLS.create_mirror_package(os.path.join(_PATH_ROOT, "src"), _PACKAGE_MAPPING)
-
-    _SETUP_MODULE = _load_py_module(name="pkg_setup", location=_PATH_SETUP)
-    _SETUP_MODULE._adjust_manifest(pkg_name=_REAL_PKG_NAME)
-    setup(**_SETUP_MODULE._setup_args(pkg_name=_REAL_PKG_NAME))
+    setup_tools = _load_py_module(name="setup_tools", location=os.path.join(".actions", "setup_tools.py"))
+
+    package_to_install = _PACKAGE_NAME or "lightning"
+    print(f"Installing the {package_to_install} package")  # requires `-v` to appear
+    if package_to_install == "lightning":
+        # install everything
+        setup_tools._load_aggregate_requirements(_PATH_REQUIRE, _FREEZE_REQUIREMENTS)
+        setup_tools.create_mirror_package(_PATH_SRC, _PACKAGE_MAPPING)
+    elif package_to_install not in _PACKAGE_MAPPING:
+        raise ValueError(f"Unexpected package name: {_PACKAGE_NAME}. Possible choices are: {list(_PACKAGE_MAPPING)}")
+
+    possible_packages = _PACKAGE_MAPPING.values() if _PACKAGE_NAME is None else [_PACKAGE_MAPPING[_PACKAGE_NAME]]
+    for pkg in possible_packages:
```

Comment:
I like it! Shall we wrap it in function, so it is easier to understand this step?

Reply:
This logic will change in followups, so I think its better to leave it unrolled for now to facilitate changes. I can add comments if a particular step is not clear